src/memory_pool.c

```diff
@@ -15,6 +15,30 @@
 #include <assert.h>
 #include <stdlib.h>
 
+enum umf_result_t
```

Comment:
Note that with recent Łukasz D. changes there is no need to use "enum" and "struct" keywords when you reference umf_result_t etc. Please remove them.